core/src/main/java/com/sequenceiq/cloudbreak/job/stackpatcher/ExistingStackPatcherConfig.java

```diff
@@ -1,40 +1,54 @@
 package com.sequenceiq.cloudbreak.job.stackpatcher;
 
+import java.util.Set;
+
 import javax.annotation.PostConstruct;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.stereotype.Component;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import com.sequenceiq.cloudbreak.domain.stack.StackPatchType;
 
-@Component
+@Configuration
+@ConfigurationProperties(prefix = "existingstackpatcher")
 public class ExistingStackPatcherConfig {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ExistingStackPatcherConfig.class);
 
-    @Value("${existingstackpatcher.intervalhours}")
     private int intervalInHours;
 
-    @Value("${existingstackpatcher.enabled}")
-    private boolean existingStackPatcherEnabled;
+    private Set<StackPatchType> enabled;
 
-    @Value("${existingstackpatcher.maxInitialStartDelayInHours}")
     private int maxInitialStartDelayInHours;
 
     @PostConstruct
     void logEnablement() {
-        LOGGER.info("Existing stack patcher is {}", existingStackPatcherEnabled ? "enabled" : "disabled");
-    }
-
-    public boolean isExistingStackPatcherEnabled() {
-        return existingStackPatcherEnabled;
+        LOGGER.info("Existing stack patchers enabled: {}", enabled);
     }
 
     public int getIntervalInHours() {
         return intervalInHours;
     }
 
+    public void setIntervalInHours(int intervalInHours) {
+        this.intervalInHours = intervalInHours;
+    }
+
+    public Set<StackPatchType> getEnabled() {
```

Comment:
Its maybe only me, but for getEnabled I'm expecting a boolean result insteand of a set of stack patch types... It would be more meaningful if you change the property to getEnabledStackPatchTypes().